src/Runner.Worker/ExecutionContext.cs

```diff
@@ -924,6 +931,47 @@ public void PublishStepTelemetry()
                 // Add to the global steps telemetry only if we have something to log.
                 if (!string.IsNullOrEmpty(StepTelemetry?.Type))
                 {
+                    if (!IsEmbedded)
+                    {
+                        StepTelemetry.Result = _record.Result;
+                    }
+
+                    if (!IsEmbedded &&
+                        _record.FinishTime != null &&
+                        _record.StartTime != null)
+                    {
+                        StepTelemetry.ExecutionTimeInSeconds = (int)Math.Ceiling((_record.FinishTime - _record.StartTime)?.TotalSeconds ?? 0);
+                    }
+
+                    if (!IsEmbedded &&
+                        _record.Issues.Count > 0)
+                    {
+                        foreach (var issue in _record.Issues)
+                        {
+                            if ((issue.Type == IssueType.Error || issue.Type == IssueType.Warning) &&
+                                !string.IsNullOrEmpty(issue.Message))
+                            {
+                                string issueTelemetry;
+                                if (issue.Message.Length > 256)
+                                {
+                                    issueTelemetry = $"{issue.Message[..256]}";
+                                }
+                                else
+                                {
+                                    issueTelemetry = issue.Message;
+                                }
+
+                                StepTelemetry.ErrorMessages.Add(issueTelemetry);
```

Comment:
we don't need to rerun secret masker since all the messages in side issues are already masked.